docs/nimble-spright-requirements-adr.md

```diff
@@ -0,0 +1,79 @@
+
+There have been cases of tension between the Nimble team and other contributing teams, because the contributing teams have felt unnecessarily slowed or burdened by the expectations and demands of the Nimble code owners. Furthermore, the burden of reviewing submissions can be significant, and can compete with the Nimble team's own commitments, especially when the contributing teams have deadline pressures. The rigor of the Nimble contribution process can become a disincentive for potential contributors and lead them to instead develop their components outside of Nimble, potentially sacrificing quality and/or reusability for development speed.
+
+Additionally, Nimble is only meant for components with general utility. Components that are product-specific for any reason do not belong in Nimble. This, too, can result in components being developed in a specific application, without the benefit of common infrastructure and established patterns, and with limited potential for reuse.
+
+Nimble contributors may opt to create their component as ["incubating"](https://github.com/ni/nimble/blob/a9ba0b6027479fe1cc2267f11957caa329910dfc/packages/nimble-components/CONTRIBUTING.md#marking-a-component-as-incubating), which comes with certain relaxed requirements but, consequently, is not considered ready for general use. It is expected that an incubating component is on the path to becoming a first-class Nimble component and will eventually meet all the necessary requirements. It may be appropriate to treat a component as incubating in cases where an MVP is useful, something temporarily blocks part of the development, or significant tech debt must be deferred to meet a deadline. Though an incubating component can have a lower initial development cost, the spec and implementation reviews may still be take considerable time/effort for the developer and the Nimble code owners.
+
+## Decision
+
+- Components with potential for reuse are strongly encouraged to go in Nimble.
+    - In some cases (mentioned in the preceeding section), the component may be added as "incubating". It should not be assumed that doing so will significantly shorten development time.
+- The [Spright repo](https://github.com/ni/spright) will be an alternative to Nimble. Ownership and review of Spright components falls to the contributing developers. It is up to those owners to define and enforce appropriate standards, but they should conform to a common set of Spright-wide minimum requirements (see table below).
```

Comment:
We should add clear notes to the decision about what the infrastructure ownership will look like in Spright. I think the current proposal is something like this:
1. separate workflows (and npm workspaces?) per contributing team
2. Nimble team maintains repo-wide infrastructure like PR status requirements, repo docs, specs

And this adds some additional consequences
1. contributing teams responsible for uptaking new Nimble versions and other Renovate PRs
2. packages in Spright might depend on differing versions of Nimble

Reply:
I wonder if we should stick with one npm workspace so that every package keeps dependency versions in sync. But if we do that, I suppose it wouldn't make sense to have separate builds. If a team wants to upgrade a dependency, it could potentially break some other packages' builds. I'm not sure whether it's more important to keep dependencies synced or have independent builds.

Reply:
In discussions with @rajsite we were really wary of the case where Nimble makes a breaking change (to a public API or an internal API that is being leveraged by a spright package). We didn't want to be responsible for uptaking those in another new repo, which means we would ask other teams to be responsible for it. But we don't trust that they would do so on a timely basis, which would mean the build would be broken for everyone if we have a monobuild. So our leaning was to move away from monobuild, which would be at the expense of keeping dependency versions in sync.

But we should think through the consequences of not keeping dependency versions in sync. Obviously we'd want versions in sync between an app->nimble and app->spright->nimble. But I think that comes for free from the way npm does dependency resolution in the app (assuming the deps are compatible). I haven't been able to think of consequences of spright libraries depending on different Nimble version. Do you have anything in mind?

Reply:
Jumping in the middle of the convo so may not be following context but for

> So our leaning was to move away from monobuild, which would be at the expense of keeping dependency versions in sync.

I don't know if I'm leaning one way or another. But I think ownership should be explicit and clear. If it's monolithic, frankly nimble would be the implicit owner so we make it explicit. If it's separate builds I think we risk lots of stagnation and inconsistencies building up and losing the ability to influence libraries moving forward (need to update Angular, blazor, foundation dependencies, etc). Also higher chance of more unconventional / wacky build configurations getting pushed through.

Reply:
I've updated the ADR based on our conversation.